File: tests/import.lua
local module = require("module")
assert(bool((module.name == "module")))
assert(bool((module.func(merge_kwargs({}, {}), 10) == 10)))
assert(bool((module.func_2(merge_kwargs({}, {}), 10) == 20)))
assert(bool((module.Module(merge_kwargs({}, {}), "first").name == "first")))
assert(bool((module.module_2.func(merge_kwargs({}, {}), 10) == 100)))
assert(bool((module.module_2.func_2(merge_kwargs({}, {}), 10) == 200)))
local new_module = require("module")
assert(bool((new_module.func(merge_kwargs({}, {}), 10) == 10)))
assert(bool((new_module.func_2(merge_kwargs({}, {}), 10) == 20)))
local module = require("module")
local new_module_2 = require("module_2")
assert(bool((module.func(merge_kwargs({}, {}), 10) == 10)))
assert(bool((module.func_2(merge_kwargs({}, {}), 10) == 20)))
assert(bool((new_module_2.func(merge_kwargs({}, {}), 10) == 100)))
assert(bool((new_module_2.func_2(merge_kwargs({}, {}), 10) == 200)))
local new_module = require("module")
local new_module_2 = require("module_2")
assert(bool((new_module.func(merge_kwargs({}, {}), 10) == 10)))
assert(bool((new_module.func_2(merge_kwargs({}, {}), 10) == 20)))
assert(bool((new_module_2.func(merge_kwargs({}, {}), 10) == 100)))
assert(bool((new_module_2.func_2(merge_kwargs({}, {}), 10) == 200)))
local func = require("module").func
local func_2 = require("module").func_2
assert(bool((func(merge_kwargs({}, {}), 10) == 10)))
assert(bool((func_2(merge_kwargs({}, {}), 10) == 20)))
local new_func = require("module").func
assert(bool((new_func(merge_kwargs({}, {}), 10) == 10)))
local func = require("module").func
local new_func_2 = require("module").func_2
assert(bool((func(merge_kwargs({}, {}), 10) == 10)))
assert(bool((new_func_2(merge_kwargs({}, {}), 10) == 20)))
local new_func = require("module").func
local new_func_2 = require("module").func_2
assert(bool((new_func(merge_kwargs({}, {}), 10) == 10)))
assert(bool((new_func_2(merge_kwargs({}, {}), 10) == 20)))
return {
    module = module,
    new_module = new_module,
    module = module,
    new_module_2 = new_module_2,
    new_module = new_module,
    new_module_2 = new_module_2,
    func = func,
    func_2 = func_2,
    new_func = new_func,
    func = func,
    new_func_2 = new_func_2,
    new_func = new_func,
    new_func_2 = new_func_2,
}

File: tests/import.py
# import

## import
import module

assert module.name == 'module'
assert module.func(10) == 10
assert module.func_2(10) == 20
assert module.Module('first').name == 'first'
assert module.module_2.func(10) == 100
assert module.module_2.func_2(10) == 200


## import as
import module as new_module

assert new_module.func(10) == 10
assert new_module.func_2(10) == 20


## import and import as
import module, module_2 as new_module_2

assert module.func(10) == 10
assert module.func_2(10) == 20

assert new_module_2.func(10) == 100
assert new_module_2.func_2(10) == 200

## multi import as
import module as new_module, module_2 as new_module_2

assert new_module.func(10) == 10
assert new_module.func_2(10) == 20

assert new_module_2.func(10) == 100
assert new_module_2.func_2(10) == 200


## from import
from module import func, func_2

assert func(10) == 10
assert func_2(10) == 20

## from import as
from module import func as new_func

assert new_func(10) == 10

## multi from import as
from module import func, func_2 as new_func_2

assert func(10) == 10
assert new_func_2(10) == 20

## multi from import as
from module import func as new_func, func_2 as new_func_2

assert new_func(10) == 10
assert new_func_2(10) == 20

